DigitalLearningSolutions.Web/Services/GroupsService.cs

```diff
@@ -601,9 +645,19 @@ private IEnumerable<Group> GetSynchronisedGroupsForCentre(int centreId)
                 .Where(g => g.ChangesToRegistrationDetailsShouldChangeGroupMembership);
         }
 
+        private IEnumerable<Group> GetAddNewRegistrantGroupsForCentre(int centreId)
+        {
+            return groupsDataService.GetGroupsForCentre(centreId)
+                .Where(g => g.ShouldAddNewRegistrantsToGroup);
+        }
+
         private static bool GroupLabelMatchesAnswer(string groupLabel, string? answer, string linkedFieldName)
         {
-            return !string.IsNullOrEmpty(answer) && string.Equals(groupLabel, answer, StringComparison.CurrentCultureIgnoreCase) || string.Equals(
+            return !string.IsNullOrEmpty(answer) && string.Equals(

```

Comment:
I should have spotted this in the earlier review: this code contains an incorrect order of operations.

`a && b || c` === `(a && b) || c`, whereas I think what we want here is `a && (b || c)`. 

As it stands, given the condition that `answer` is an empty string and `groupLabel` is `{linkedFieldName} - `, this function will return `true`, which would _probably_ be surprising. Obviously we know this situation is not going to exist, but I generally consider it code-smell to have a mixture of `&&` and `||` without brackets to explicitly show what is happening. 

As an added bonus, changing this to `a && (b || c)` makes it slightly more optimized: currently if `answer` is null or empty string, the second `string.Equals` gets executed, and as it _should_ be impossible for this to return true (given sensible values for `groupLabel` in the database), this is an unnecessary operation.